### PR TITLE
Sync: master -> grpc

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

File: core/base/src/main/java/alluxio/uri/Authority.java [hunk 1]
```diff
@@ -45,6 +46,10 @@ static Authority fromString(String authority) {
       if (matcher.matches()) {
         return new SingleMasterAuthority(matcher.group(1), Integer.parseInt(matcher.group(2)));
       } else {
+        matcher = MULTI_MASTERS_AUTH.matcher(authority);
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply core/base/src/main/java/alluxio/uri/Authority.java]: The master branch one is correct.
`MultiMasterAuthority(authority.replaceAll("[;+]", ","));`

**Comment by LuQQiu** [comment]
@gpang The multi-master authority change is illustrated in https://github.com/Alluxio/alluxio/pull/8237#discussion_r245766007

But feel free to directly merge it (without the newest master branch multi master authority changes) and I can change them back in later PR.

File: core/common/src/test/java/alluxio/AlluxioURITest.java [hunk 2]
```diff
@@ -121,6 +122,53 @@ public void basicTwoPartUri() {
...
+        uri.join(new AlluxioURI("/d")).toString());
+    assertEquals("alluxio://host1:19998,host2:19998,host3:19998/xy z/a b c", uri.toString());
+  }
+
+  @Test
+  public void semicolonMultiMasterUri() {
+    AlluxioURI uri =
+        new AlluxioURI("alluxio://host1:1323;host2:54325;host3:64354/xy z/a b c");
+    assertTrue(uri.hasAuthority());
+    assertEquals("host1:1323;host2:54325;host3:64354", uri.getAuthority().toString());
```
[the remark this answers is not in the document]
Comment (ggezer) [reply core/common/src/test/java/alluxio/AlluxioURITest.java]: I don't have this change at the merge time.

File: core/common/src/main/java/alluxio/grpc/GrpcExceptionUtils.java [hunk 3]
```diff
@@ -100,11 +100,11 @@ public static AlluxioStatusException fromGrpcStatusException(StatusRuntimeExcept
         alluxioStatus = alluxio.exception.status.Status.UNKNOWN;
     }
 
-    Throwable cause = (e.getCause() != null) ? e.getCause() : e;
+    Throwable innerMostCause = (e.getCause() != null) ? e.getCause() : e;
```
[the remark this answers is not in the document]
Comment (ggezer) [reply core/common/src/main/java/alluxio/grpc/GrpcExceptionUtils.java]: gRPC can't chain exceptions. We have to simulate that. I agree "innerMost" does not make sense here and changed it. Also added comments in appropriate places.

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
@gpang Removed the thriftGen as well.

**Pull request merged**